### DISCUSSION TITLE
How secure is the user data from admin access?

### DISCUSSION THREAD

**Discussion by Bockiii** [dani-garcia/vaultwarden#1655] [Q&A] [answered]
Greetings,

first off: Thanks a lot for this! I was so bummed to find that even the self hosted original container doesn't allow totp support, I was half-way switching to keepass before I found this. So: Thanks a lot!

Now to the question: If I want to give access to friends and family, how can I assure them that I don't have access to their data? Is there a disclaimer somewhere explaining how data is stored/secured to prevent leaks, even to the admin/hoster?

I wouldn't access the data anyways, but especially with credit card infos and passwords, proof is better than promise :)

Thanks in advance!

**Comment by dani-garcia** [comment, marked as answer]
Any sensitive data is encrypted by the clients before it's sent to the server, you can read more in the official help page: https://bitwarden.com/help/article/what-encryption-is-used/

Vaultwarden works in exactly the same way as the official server, I think the only unencrypted data you really have access to is their email addresses and names, because it's needed for logins and sending emails.

**Reply by Bockiii** [reply]
Thanks for the answer!

It's exactly what I had hoped for :)

Greetings
